### PR TITLE
KAFKA-17650: validate ReplaceField config

### PR DESCRIPTION
**Pull request by TaiJuWu**
jira: https://issues.apache.org/jira/browse/KAFKA-17650
There is no validation of ReplaceField, we should add it.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java [hunk 1]
```diff
@@ -109,6 +117,14 @@ public void configure(Map<String, ?> configs) {
         schemaUpdateCache = new SynchronizedCache<>(new LRUCache<>(16));
     }
 
+    private static void vilidate(Map<String, ?> configs) {
+        for (Map.Entry<String, ?> entry : configs.entrySet()) {
+            if (!names().contains(entry.getKey())) {
+                throw new InvalidConfigurationException("Unknown config: " + entry.getKey());
+            }
+        }
```
Comment (m1a2st) [inline connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java]: how about 
```java
configs.keySet().forEach(key -> {
        if (!names().contains(key)) {
            throw new InvalidConfigurationException("Unknown config: " + key);
        }
    });
```

File: connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java [hunk 2]
```diff
@@ -95,6 +100,9 @@ public String version() {
 
     @Override
     public void configure(Map<String, ?> configs) {
+
+        vilidate(configs);
```
Comment (m1a2st) [inline connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java]: This is a typo about `vilidate`, should be `validate`

File: connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java [hunk 3]
```diff
@@ -78,6 +79,10 @@ interface ConfigName {
             .define(ConfigName.REPLACE_NULL_WITH_DEFAULT_CONFIG, ConfigDef.Type.BOOLEAN, true, ConfigDef.Importance.MEDIUM,
                     "Whether to replace fields that have a default value and that are null to the default value. When set to true, the default value is used, otherwise null is used.");
 
+    private static Set<String> names() {
```
Comment (m1a2st) [inline connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java]: nit: I think it is more clearly to use `configNames`

**Review by m1a2st — COMMENTED** [review]
@TaiJuWu, Thanks for your PR, I left some comments, PTAL

File: connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java (hunk 3 above)
Comment (TaiJuWu) [reply connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java]: Done. Thanks for review.

**Review by m1a2st — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
@TaiJuWu Could you please file a jira as it is a kind of bug

**Comment by TaiJuWu** [comment]
> @TaiJuWu Could you please file a jira as it is a kind of bug

Done. Updated in title and post, thanks!

**Comment by chia7712** [comment]
@TaiJuWu why we need this changes? the other fields have no such checks.

**Comment by TaiJuWu** [comment]
> @TaiJuWu why we need this changes? the other fields have no such checks.

I noticed the check in the client-side module and believe it could also be applied to the connect module. 
Personally, I would prefer that all user-provided configurations are explicitly defined. 
However, if this check is unnecessary, I’m happy to close the PR.

**Comment by chia7712** [comment]
> Personally, I would prefer that all user-provided configurations are explicitly defined.

that is a kind of behavior change. We need KIP if it is necessary improvement.

File: connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java [hunk 4]
```diff
@@ -72,6 +73,10 @@ interface ConfigName {
             .define(ConfigName.REPLACE_NULL_WITH_DEFAULT_CONFIG, ConfigDef.Type.BOOLEAN, true, ConfigDef.Importance.MEDIUM,
                     "Whether to replace fields that have a default value and that are null to the default value. When set to true, the default value is used, otherwise null is used.");
 
+    private static Set<String> configNames() {
```
Comment (AndrewJSchofield) [inline connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java:76]: I note that `ConfigName.RENAME` actually has the value `"renames"` so it probably should be `ConfigName.RENAMES`. This probably ought to be resolved.

Comment (chia7712) [reply connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java:76]: @TaiJuWu Maybe you can file a minor PR to fix the naming. this PR needs more discussion due to behavior change.

Comment (TaiJuWu) [reply connect/transforms/src/main/java/org/apache/kafka/connect/transforms/ReplaceField.java:76]: > @TaiJuWu Maybe you can file a minor PR to fix the naming. this PR needs more discussion due to behavior change.

Will do, thanks for helping check.

**Pull request closed without merging**